src/rust/ide/view/graph-editor/src/component/node/output/port.rs

```diff
@@ -109,6 +107,24 @@ impl AllPortsShape {
     }
 }
 
+fn shape_border_length<T>(corner_radius: T, width: T) -> T
```

Comment:
please add docs to these functions - they are not obvious. For example, what is "border" here? And what is border length? In the next function, what is "segment" - these should be described in docs.